### PR TITLE
Add onContentVisible Param

### PR DESCRIPTION
**Pull request by reintroducing**
Adding an optional callback method that you can call when the content becomes visible.

### CODE REVIEW

File: src/LazyLoad.jsx [hunk 1]
```diff
@@ -60,10 +64,11 @@ LazyLoad.propTypes = {
   children: PropTypes.node.isRequired,
   height: PropTypes.oneOfType([
     PropTypes.string,
-    PropTypes.number,
```
Comment (sergeylaptev) [inline src/LazyLoad.jsx]: Why you deleted trailing commas?:)

Comment (reintroducing) [inline src/LazyLoad.jsx]: lol, i know IE8 doesn't like it, not that i care about IE8 anymore but it's a habit :)

Comment (sergeylaptev) [inline src/LazyLoad.jsx]: Babel automatically removes trailing commas https://github.com/airbnb/javascript/issues/323

Comment (reintroducing) [inline src/LazyLoad.jsx]: Ok, no problem, I can remove it. Other than that, this look ok to merge after I put the commas back?

Comment (sergeylaptev) [inline src/LazyLoad.jsx]: There is no reason for reject it:)

File: src/LazyLoad.jsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ export default class LazyLoad extends Component {
     window.removeEventListener('resize', this.onWindowScroll);
   }
   onWindowScroll() {
-    const { threshold } = this.props;
+    const { threshold, onContentVisible } = this.props;
```
Comment (sergeylaptev) [inline src/LazyLoad.jsx]: Need to change order of props, { onContentVisible, threshold }

Comment (reintroducing) [inline src/LazyLoad.jsx]: I can do that but just curious why, alphabetical preference?

**Comment by reintroducing** [comment]
Requested updates made.

**Comment by sergeylaptev** [comment]
Alphabetical order for props and destructuring them in accordance with https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md
and
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-prop-types.md
I've just not implented eslint yet. Anyway than less changes in future than better:)

File: src/LazyLoad.jsx [hunk 3]
```diff
@@ -35,7 +35,11 @@ export default class LazyLoad extends Component {
 
     if (top === 0 || (top <= (scrollTop + window.innerHeight + threshold)
                       && (top + height) > (scrollTop - threshold))) {
-      this.setState({ visible: true });
+      this.setState({ visible: true }, () => {
+        if (onContentVisible) {
```
Comment (sergeylaptev) [inline src/LazyLoad.jsx]: Maybe will be better move it into onVisible method? Using callback of setState isn't good idea in most cases

**Comment by sergeylaptev** [comment]
Thanks for your work, I'll refactor some things later on.

**Comment by reintroducing** [comment]
Sure thing, just wanted to get it in there for use, I expected you'd refactor to your liking.

File: src/LazyLoad.jsx (hunk 3 above)
Comment (reintroducing) [inline src/LazyLoad.jsx]: I'm relatively new to React and have not heard of using the callback of setState as being a bad idea. Could you give a bit of insight into why this is such?

Comment (sergeylaptev) [inline src/LazyLoad.jsx]: This is not declarative way. There is difference between manually triggering event callback and listening changes, comparing current and previous states and trigger callback when they have been changed. In second case we have 2 decoupled parts. The part which is responsible for update visible state shouldn't know about triggering some events outside, but if you want you can dedicate this logic in an another place (didUpdate hook)

Comment (reintroducing) [inline src/LazyLoad.jsx]: Ah you're absolutely right, I actually just saw that convention used in another module and thought it was better than what I was doing. Thanks for the explanation.

Comment (sergeylaptev) [inline src/LazyLoad.jsx]: You are welcome!;)

**Pull request merged**